driver-core/src/main/com/mongodb/internal/connection/PowerOfTwoBufferPool.java

```diff
@@ -34,9 +38,64 @@
  *
  * <p>This class should not be considered a part of the public API.</p>
  */
-public class PowerOfTwoBufferPool implements BufferProvider {
+public class PowerOfTwoBufferPool implements BufferProvider, AutoCloseable {
 
-    private final Map<Integer, ConcurrentPool<ByteBuffer>> powerOfTwoToPoolMap = new HashMap<Integer, ConcurrentPool<ByteBuffer>>();
+    /**
+     * The global default pool.  Pruning is enabled on this pool. Idle buffers are pruned after one minute.
+     */
+    public static final PowerOfTwoBufferPool DEFAULT = new PowerOfTwoBufferPool();
+
+    private static final long DEFAULT_MAX_IDLE_TIME_MINUTES = 1;
+
+    static {
+        DEFAULT.enablePruning();
+    }
+
+    private static final class IdleTrackingByteBuffer {
+        private final long lastUsedNanos;
+        private final ByteBuffer buffer;
+
+        private IdleTrackingByteBuffer(final ByteBuffer buffer) {
+            this.lastUsedNanos = System.nanoTime();
+            this.buffer = buffer;
+        }
+
+        public long getLastUsedNanos() {
+            return lastUsedNanos;
+        }
+
+        public ByteBuffer getBuffer() {
+            return buffer;
+        }
+    }
+
+    private final class ItemFactory implements ConcurrentPool.ItemFactory<IdleTrackingByteBuffer> {
+        private final int size;
+
+        private ItemFactory(final int size) {
+            this.size = size;
+        }
+
+        @Override
+        public IdleTrackingByteBuffer create() {
+            return new IdleTrackingByteBuffer(createNew(size));
+        }
+
+        @Override
+        public void close(final IdleTrackingByteBuffer idleTrackingByteBuffer) {
+        }
+
+        @Override
+        public Prune shouldPrune(final IdleTrackingByteBuffer idleTrackingByteBuffer) {
+            return idleTrackingByteBuffer.getLastUsedNanos() < System.nanoTime() + maxIdleTimeNanos
+                    ? Prune.YES : Prune.STOP;
```

Comment:
Stop when we find the first non-idle buffer, since they should be roughly ordered in the pool by last use time (though no lock strictly enforces that invariant)